trick_source/sim_services/MonteCarlo/MonteVarFile.cpp

```diff
@@ -34,6 +34,17 @@ std::string Trick::MonteVarFile::describe_variable() {
 }
 
 std::string Trick::MonteVarFile::get_next_value() {
+
+    // Reopen the file if not open.
+    if(!input_file_stream->is_open()) {
+        input_file_stream->open(this->file_name.c_str(), std::ifstream::in);
+
+        // If the stream position has been set, re-open file to this position.
+        if(this->stream_position != NULL) {
```

Comment:
This comparison should be made against 0 instead of NULL.  Although NULL will most likely work here, NULL is usually used when comparing pointer values.  The stream_position is most likely a large integer type.

Reply:
I see. It's different in Java and C#. I'll make the change real quick.

Reply:
This check is not necessary. `stream_position` should be initialized to `0` in the constructor. Thereafter, it is set only via `tellg()` in this function. It is always valid.

Reply:
Correct. Another redundant check. I can remove it.